buildSrc/src/main/kotlin/configureMavenSigning.kt

```diff
@@ -0,0 +1,10 @@
+import org.gradle.kotlin.dsl.get
+import org.gradle.plugins.signing.SigningExtension
+
+fun SigningExtension.addMavenCentralSigning() {
+    val gpgBase64Key = project.providers.gradleProperty("sonatypePgpKeyBase64").forUseAtConfigurationTime()
```

Comment:
I think it's best to use a `base64` formatted GPG key as that can be most easily injected into Gradle as a property. Should you have different requirements, then this `SigningExtension` also need to be modified.